test/uk/gov/hmrc/transitmovementsrouter/services/StreamingMessageTrimmerSpec.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.transitmovementsrouter.services
+
+import akka.stream.scaladsl.Sink
+import akka.stream.scaladsl.Source
+import akka.util.ByteString
+import org.scalatest.concurrent.ScalaFutures
+import org.scalatest.freespec.AnyFreeSpec
+import org.scalatest.matchers.must.Matchers
+import uk.gov.hmrc.transitmovementsrouter.base.TestActorSystem
+
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.reflect.runtime.universe.Try
+import scala.util.Failure
+import scala.util.Success
+
+class StreamingMessageTrimmerSpec extends AnyFreeSpec with Matchers with ScalaFutures with TestActorSystem {
+  "trim flow should" - {
+
+    val sut = new StreamingMessageTrimmerImpl
+
+    "successfully trim with valid xml" in {
+      val input = "<TraderChannelResponse><test>text</test></TraderChannelResponse>"
```

Comment:
Is TraderChannelResponse going to be the first element in the message? I don't think the subslice will work if it's embedded, in that case you'd need to provide the path from the root node

Reply:
It will be - EIS wrap the XML and this should be the root element.